Cloudflare build fails due to Node APIs

Importing only `createContainer` from awilix causes the build to crash.

It seems that `listModules` (which relies on `fast-glob`) is being implicitly included, **even when not used** which is a problem as Cloudflare Pages and Workers have limited support for Node.js APIs (with `nodejs_compat`).

---

Workaround : import from the browser build and manually assign the types
```js
import type { asFunction as asFunctionTyped, createContainer as createContainerTyped } from 'awilix'

// @ts-expect-error
import { asFunction as _asFunction, createContainer as _createContainer } from 'awilix/lib/awilix.browser.js'

const asFunction: typeof asFunctionTyped = _asFunction
const createContainer: typeof createContainerTyped = _createContainer
```

Without `awilix` (no browser) I get these build errors locally with adapter-cloudflare:
```
> Using @sveltejs/adapter-cloudflare
✘ [ERROR] Could not resolve "fs"

    node_modules/@nodelib/fs.scandir/out/adapters/fs.js:4:19:
      4 │ const fs = require("fs");
        │                    ~~~~
        ╵                    "./fs"

  Cannot use "fs" when deploying to Cloudflare.


✘ [ERROR] Could not resolve "fs"

    node_modules/@nodelib/fs.stat/out/adapters/fs.js:4:19:
      4 │ const fs = require("fs");
        │                    ~~~~
        ╵                    "./fs"

  Cannot use "fs" when deploying to Cloudflare.


✘ [ERROR] Could not resolve "url"

    node_modules/awilix/lib/awilix.module.mjs:6:30:
      6 │ import { pathToFileURL } from 'url';
        ╵                               ~~~~~

  Cannot use "url" when deploying to Cloudflare.


✘ [ERROR] Could not resolve "os"

    node_modules/fast-glob/node_modules/glob-parent/index.js:5:22:
      5 │ var isWin32 = require('os').platform() === 'win32';
        ╵                       ~~~~

  Cannot use "os" when deploying to Cloudflare.


✘ [ERROR] Could not resolve "fs"

    node_modules/fast-glob/out/settings.js:4:19:
      4 │ const fs = require("fs");
        ╵                    ~~~~

  Cannot use "fs" when deploying to Cloudflare.


✘ [ERROR] Could not resolve "os"

    node_modules/fast-glob/out/settings.js:5:19:
      5 │ const os = require("os");
        ╵                    ~~~~

  Cannot use "os" when deploying to Cloudflare.


✘ [ERROR] Could not resolve "os"

    node_modules/fast-glob/out/utils/path.js:4:19:
      4 │ const os = require("os");
        ╵                    ~~~~

  Cannot use "os" when deploying to Cloudflare.


error during build:
Error: Bundling with esbuild failed with 7 errors
```

I thought as much :+1: 
Different error now (local build):
```ts
      '.': {
        import: './lib/awilix.module.mjs',
        types: './lib/awilix.d.ts',
        default: './lib/awilix.js',
        workerd: './lib/awilix.browser.js',
      },
```
results in
```
file://node_modules/awilix/lib/awilix.module.mjs:4
import { importModule } from './load-module-native.js';
         ^^^^^^^^^^^^
SyntaxError: The requested module './load-module-native.js' does not provide an export named 'importModule'
```
And in CF it fails with the other Node import errors (e.g. `✘ [ERROR] Could not resolve "fs"`)

Nope, `import { ... } from 'awilix'` doesn't work, I end up with: 
```
> Using @sveltejs/adapter-cloudflare
✘ [ERROR] Could not resolve "fs"
```
and a bunch of other similar errors.

--- a/README.md
+++ b/README.md
@@ -1384,7 +1384,7 @@ The package includes 4 flavors.
 - CommonJS, the good ol' Node format - `lib/awilix.js`
 - ES Modules, for use with module bundlers **in Node** - `lib/awilix.module.mjs`
 - ES Modules, for use with module bundlers **in the browser** -
-  `lib/awilix.browser.js`
+  `lib/awilix.browser.mjs`
 - UMD, for dropping it into a script tag - `lib/awilix.umd.js`
 
 The `package.json` includes the proper fields for bundlers like Webpack, Rollup

--- a/package.json
+++ b/package.json
@@ -5,9 +5,9 @@
   "main": "lib/awilix.js",
   "module": "lib/awilix.module.mjs",
   "jsnext:main": "lib/awilix.module.mjs",
-  "browser": "lib/awilix.browser.js",
+  "browser": "lib/awilix.browser.mjs",
   "umd:main": "lib/awilix.umd.js",
-  "react-native": "lib/awilix.browser.js",
+  "react-native": "lib/awilix.browser.mjs",
   "typings": "lib/awilix.d.ts",
   "engines": {
     "node": ">=16.3.0"
@@ -16,12 +16,19 @@
     ".": {
       "import": "./lib/awilix.module.mjs",
       "types": "./lib/awilix.d.ts",
+      "workerd": "./lib/awilix.browser.mjs",
+      "browser": {
+        "import": "./lib/awilix.browser.mjs",
+        "default": "./lib/awilix.umd.js"
+      },
       "default": "./lib/awilix.js"
     },
     "./browser": {
-      "import": "./lib/awilix.browser.js",
+      "import": "./lib/awilix.browser.mjs",
       "types": "./lib/awilix.d.ts",
-      "default": "./lib/awilix.browser.js"
+      "workerd": "./lib/awilix.browser.mjs",
+      "browser": "./lib/awilix.browser.mjs",
+      "default": "./lib/awilix.umd.js"
     },
     "./lib/*.js": {
       "types": "./lib/*.d.ts",
@@ -39,7 +46,7 @@
     "cover": "npm run test -- --coverage",
     "publish:pre": "npm run lint && npm run build && npm run cover",
     "publish:post": "npm publish && git push --follow-tags",
-    "release:prerelease": "npm run publish:pre && npm version prerelease && npm run publish:post",
+    "release:prerelease": "npm run publish:pre && npm version prerelease --preid alpha && npm run publish:post",
     "release:patch": "npm run publish:pre && npm version patch && npm run publish:post",
     "release:minor": "npm run publish:pre && npm version minor && npm run publish:post",
     "release:major": "npm run publish:pre && npm version major && npm run publish:post",

--- a/rollup.config.mjs
+++ b/rollup.config.mjs
@@ -58,7 +58,7 @@ export default [
     output: [
       {
         name: 'Awilix',
-        file: 'lib/awilix.browser.js',
+        file: 'lib/awilix.browser.mjs',
         format: 'es',
       },
       {

--- a/src/__tests__/rollup.test.ts
+++ b/src/__tests__/rollup.test.ts
@@ -4,7 +4,7 @@ const cjs = require('../../lib/awilix')
 // @ts-ignore
 import * as es from '../../lib/awilix.module.mjs'
 const umd = require('../../lib/awilix.umd')
-const browser = require('../../lib/awilix.browser')
+const browser = require('../../lib/awilix.browser.mjs')
 
 describe('rollup artifacts', () => {
   it('works', () => {

